src/vma/sock/sockinfo_udp.cpp

```diff
@@ -1775,16 +1785,45 @@ bool sockinfo_udp::tx_check_if_would_not_block()
     #pragma BullseyeCoverage on
 #endif
 
-bool sockinfo_udp::rx_input_cb(mem_buf_desc_t* p_desc, void* pv_fd_ready_array)
+/**
+ * sockinfo_udp::inspect_uc_packet inspects the input packet for basic rules,
+ * common for all cases. Its applicable for UC case as well.
+ */
+inline bool sockinfo_udp::inspect_uc_packet(mem_buf_desc_t* p_desc)
 {
 	// Check that sockinfo is bound to the packets dest port
+	// It is prohibiting to receive packets in case of closing and fast reopening socket.
+	// In such case bind() will assign new port and old packets still in CQ will be dropped
+	// here for both FlowTag and usual case. Note if bind() would forced with same as 
+	// previous port # user app may receive wrong packets depending on CQ state.
```

Comment:
I propose the following comment:

```
// This protects the case where a socket is closed and a new one is rapidly opened
// receiving the same socket id.
// In this case packets arriving for the old sockets should be dropped.
// This distinction assumes that the OS guarentees the old and new sockets to recieve different
// port numbers from bind().
// If the user requests to bind the new socket to the same port number as the old one it will be
// impossible to identify packets desginated for the old socket in this way.
```